src/plugins/csvstorage/csvstorage.c

```diff
@@ -0,0 +1,350 @@
+{
+	char *line = (_line + offset);
+	if(*line == '\0')
+		return NULL;
+	char *ptr = strchr(line, delim);
+
+	if(ptr == NULL)
+	{
+		line[strlen(line)-1]='\0';
+	}
+	else
+		*ptr = '\0';
+	return line;
+}
+
+static unsigned long getLineLength(FILE *fp)
+{
+	int startPos = ftell(fp);
+	char c;
+	while((c = fgetc(fp)) && (!feof(fp)))
+	{
+		if(c == '\n')
+			break;
+	}
+	int  endPos = ftell(fp);
+	fseek(fp, startPos, SEEK_SET);
+	if((endPos - startPos) == 0)
+		return 0;
+	else
+		return (endPos - startPos)+1;
+}
+
+static unsigned long getColumnCount(char *lineBuffer, char delim)
+{
+	char *ptr = lineBuffer;
+	unsigned long counter = 0;
+	while(*ptr != '\0')
+	{
+		if(*ptr == delim)
+			++counter;
+		++ptr;
+	}
+	++counter;
+	return counter;
+}
+
+char *itostr(char *buf, unsigned long i, uint8_t len)
+{
+	snprintf(buf, len, "%lu", i);
+	return buf;
+}
+
+static Key *getKeyByOrderNr(KeySet *ks, unsigned long n)
+{
+	Key *cur;
+	char buf[16];
+	ksRewind(ks);
+	while((cur = ksNext(ks)) != NULL)
+	{
+		if(!strcmp(keyString(keyGetMeta(cur, "csvOrder")), itostr(buf, n, sizeof(buf)-1)))
+			return cur; 
+	}
+	return NULL;
+}
+int elektraCsvstorageGet(Plugin *handle, KeySet *returned, Key *parentKey)
+{
+	if (!strcmp(keyName(parentKey), "system/elektra/modules/csvstorage"))
+	{
+		KeySet *contract = ksNew (30,
+				keyNew ("system/elektra/modules/csvstorage",
+					KEY_VALUE, "csvstorage plugin waits for your orders", KEY_END),
+				keyNew ("system/elektra/modules/csvstorage/exports", KEY_END),
+				keyNew ("system/elektra/modules/csvstorage/exports/get",
+					KEY_FUNC, elektraCsvstorageGet, KEY_END),
+				keyNew ("system/elektra/modules/csvstorage/exports/set",
+					KEY_FUNC, elektraCsvstorageSet, KEY_END),
+#include ELEKTRA_README(csvstorage)
+				keyNew ("system/elektra/modules/csvstorage/infos/version",
+					KEY_VALUE, PLUGINVERSION, KEY_END),
+				KS_END);
+		ksAppend (returned, contract);
+		ksDel (contract);
+
+		return 1; /* success */
+	}
+	/* get all keys */
+	KeySet *config = elektraPluginGetConfig(handle);
+	Key *delimKey = ksLookupByName(config, "/delimiter", 0);
+	Key *readHeaderKey = ksLookupByName(config, "/useheader", 0);
+	Key *key;
+	KeySet *header = ksNew(0, KS_END);
+	Key *dirKey;
+	Key *cur;
+	FILE *fp = NULL;
+	char *col;
+	char buf[15];
+	int nr_keys = 1;
+	
+	char delim = ';';
+	if(delimKey)
+		delim = ((char *)keyString(delimKey))[0];
+	
+	uint8_t useHeader = 0;
+	if(readHeaderKey)
+	{
+		if((((char *)keyString(readHeaderKey))[0] - '0') == 1)
+			useHeader = 1;
+	}
+	const char *fileName;
+	fileName = keyString(parentKey);
+	fp = fopen(fileName, "rb");
+	if(!fp)
+	{
+		ELEKTRA_SET_ERROR(116, parentKey, "couldn't open file");
+		return -1;
+	}
+
+	unsigned long length = 0;
+	length = getLineLength(fp);
+	if(length == 0)
+	{
+		ELEKTRA_ADD_WARNING(118, parentKey, "Empty file");
+		fclose(fp);
+		return 1;
+	}
+
+	char *lineBuffer;
+	lineBuffer = malloc(length * sizeof(char)+1);
```

Comment:
please use `elektraMalloc` instead of `malloc`. Its useful for profiling and memory allocation tests.